### PR TITLE
Validate deserialization of flag data

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: controller-api/src/test/resources/system-flags/flags/flag-with-empty-data/main.controller.json [hunk 1]
```diff
@@ -1,4 +1,3 @@
 {
-  "id" : "flag-with-empty-data",
-  "comment": "empty data using only id field"
```
Comment (freva) [inline controller-api/src/test/resources/system-flags/flags/flag-with-empty-data/main.controller.json:3]: How about we just ignore all key-pairs if key == "comment"?

**Review by freva — APPROVED** [review]

File: vespajlib/src/main/java/com/yahoo/text/JSON.java [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import java.util.Map;
+import java.util.Objects;
+
+import static com.yahoo.yolean.Exceptions.uncheck;
 
 /**
- * Static methods for working with the map textual format which is parsed by {@link MapParser}
```
Comment (bjorncs) [inline vespajlib/src/main/java/com/yahoo/text/JSON.java:7]: "The text format which name should not be spoken" 🤔

File: vespajlib/pom.xml [hunk 3]
```diff
@@ -38,6 +38,11 @@
       <artifactId>guava</artifactId>
       <scope>provided</scope>
     </dependency>
+    <dependency>
```
Comment (bjorncs) [inline vespajlib/pom.xml:41]: Note: Several modules uses `vespajlib` in compile scope - some bundles may require additional Jackson packages.

**Review by bjorncs — APPROVED** [review]

**Review by freva — APPROVED** [review]

File: vespajlib/src/main/java/com/yahoo/text/JSON.java (hunk 2 above)
Comment (hakonhall) [reply vespajlib/src/main/java/com/yahoo/text/JSON.java:7]: what do you mean? the name is "map textual format", but this class is now more than just utils for that.

File: vespajlib/pom.xml (hunk 3 above)
Comment (hakonhall) [reply vespajlib/pom.xml]: Good point, I have removed the dependency and use Slime instead.

I have discussed the Slime changes with @havardpe .

File: vespajlib/src/main/java/com/yahoo/text/JSON.java (hunk 2 above)
Comment (bjorncs) [reply vespajlib/src/main/java/com/yahoo/text/JSON.java:7]: Would be simpler with "Static methods for working with JSON" ;)

**Review by bjorncs — APPROVED** [review]

**Pull request merged**